undefined reference when building libwacom 2.19.0 with glib 2.78

```
2026-06-01 01:24:19 FAILED: [code=1] libwacom.so.9.0.0 
2026-06-01 01:24:19 cc  -o libwacom.so.9.0.0 libwacom.so.9.0.0.p/libwacom_libwacom.c.o libwacom.so.9.0.0.p/libwacom_libwacom-error.c.o libwacom.so.9.0.0.p/libwacom_libwacom-database.c.o -Wl,--as-needed -Wl,--no-undefined -shared -fPIC -Wl,-soname,libwacom.so.9 -Wl,-z,relro -Wl,-z,now -specs=/usr/lib/rpm/generic-hardened-ld -O2 -g -grecord-gcc-switches -pipe -fstack-protector-strong -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/generic-hardened-cc1 -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -Wl,--version-script,/home/lkp/rpmbuild/BUILD/libwacom-2.19.0/libwacom/libwacom.sym -Wl,--start-group /usr/lib64/libgudev-1.0.so /usr/lib64/libgobject-2.0.so /usr/lib64/libglib-2.0.so /usr/lib64/libevdev.so -Wl,--end-group
2026-06-01 01:24:19 /usr/bin/ld: libwacom.so.9.0.0.p/libwacom_libwacom.c.o: in function `print_styli_for_device':
2026-06-01 01:24:19 /home/lkp/rpmbuild/BUILD/libwacom-2.19.0/x86_64-openEuler-linux-gnu/../libwacom/libwacom.c:945:(.text+0x2a77): undefined reference to `g_strv_builder_take'
2026-06-01 01:24:19 /usr/bin/ld: libwacom.so.9.0.0.p/libwacom_libwacom.c.o: in function `print_button_evdev_codes':
2026-06-01 01:24:19 /home/lkp/rpmbuild/BUILD/libwacom-2.19.0/x86_64-openEuler-linux-gnu/../libwacom/libwacom.c:1056:(.text+0x2f4a): undefined reference to `g_strv_builder_take'
2026-06-01 01:24:19 /usr/bin/ld: libwacom.so.9.0.0.p/libwacom_libwacom-database.c.o: in function `stylus_ids_as_hex':
2026-06-01 01:24:19 /home/lkp/rpmbuild/BUILD/libwacom-2.19.0/x86_64-openEuler-linux-gnu/../libwacom/libwacom-database.c:430:(.text+0x1371): undefined reference to `g_strv_builder_take'
2026-06-01 01:24:19 collect2: error: ld returned 1 exit status
```

It seems that `g_strv_builder_take` is [only available since 2.80](https://docs.gtk.org/glib/method.StrvBuilder.take.html). May glib requirments should be bumped into 2.80, or the call to `g_strv_builder_take` may be tweaked?

--- a/libwacom/libwacom-database.c
+++ b/libwacom/libwacom-database.c
@@ -427,7 +427,8 @@ stylus_ids_as_hex(GArray *array)
 
 	for (guint i = 0; i < array->len; i++) {
 		id = &g_array_index(array, WacomStylusId, i);
-		g_strv_builder_take(builder, g_strdup_printf("0x%08x", id->tool_id));
+		g_autofree char *hex = g_strdup_printf("0x%08x", id->tool_id);
+		g_strv_builder_add(builder, hex);
 	}
 
 	return g_strv_builder_end(builder);

--- a/libwacom/libwacom.c
+++ b/libwacom/libwacom.c
@@ -934,7 +934,7 @@ print_styli_for_device(int fd,
 	styli = libwacom_get_styli(device, &nstyli);
 	for (int i = 0; i < nstyli; i++) {
 		const WacomStylus *stylus = styli[i];
-		char *s;
+		g_autofree char *s = NULL;
 
 		if (stylus->id.vid != WACOM_VENDOR_ID)
 			s = g_strdup_printf("0x%04x:%#x",
@@ -942,7 +942,7 @@ print_styli_for_device(int fd,
 					    stylus->id.tool_id);
 		else
 			s = g_strdup_printf("%#x", stylus->id.tool_id);
-		g_strv_builder_take(builder, s);
+		g_strv_builder_add(builder, s);
 	}
 
 	strv = g_strv_builder_end(builder);
@@ -1046,14 +1046,14 @@ print_button_evdev_codes(int fd,
 	for (char b = 'A'; b < 'A' + nbuttons; b++) {
 		unsigned int code = libwacom_get_button_evdev_code(device, b);
 		const char *name = libevdev_event_code_get_name(EV_KEY, code);
-		char *s;
+		g_autofree char *s = NULL;
 
 		if (name)
 			s = g_strdup(name);
 		else
 			s = g_strdup_printf("0x%x", code);
 
-		g_strv_builder_take(builder, s);
+		g_strv_builder_add(builder, s);
 	}
 
 	strv = g_strv_builder_end(builder);

--- a/tools/debug-device.c
+++ b/tools/debug-device.c
@@ -428,8 +428,9 @@ handle_device(WacomDeviceDatabase *db,
 			g_auto(GStrv) strv = NULL;
 			g_autofree char *str = NULL;
 			for (int i = 0; i < nstyli; i++) {
-				char *s = g_strdup_printf("0x%06x", styli[i]);
-				g_strv_builder_take(builder, s);
+				g_autofree char *s =
+					g_strdup_printf("0x%06x", styli[i]);
+				g_strv_builder_add(builder, s);
 			}
 			strv = g_strv_builder_end(builder);
 			str = g_strjoinv(", ", strv);
@@ -447,11 +448,11 @@ handle_device(WacomDeviceDatabase *db,
 			g_autofree char *str = NULL;
 			for (int i = 0; i < nstyli; i++) {
 				const WacomStylus *s = styli[i];
-				char *entry = g_strdup_printf(
+				g_autofree char *entry = g_strdup_printf(
 					"[0x%04x, 0x%06x]",
 					libwacom_stylus_get_vendor_id(s),
 					libwacom_stylus_get_id(s));
-				g_strv_builder_take(builder, entry);
+				g_strv_builder_add(builder, entry);
 			}
 			strv = g_strv_builder_end(builder);
 			str = g_strjoinv(", ", strv);
@@ -518,9 +519,10 @@ handle_device(WacomDeviceDatabase *db,
 #pragma GCC diagnostic pop
 
 					for (int i = 0; i < npaired; i++) {
-						char *s = g_strdup_printf("0x%06x",
-									  paired[i]);
-						g_strv_builder_take(builder, s);
+						g_autofree char *s =
+							g_strdup_printf("0x%06x",
+									paired[i]);
+						g_strv_builder_add(builder, s);
 					}
 					strv = g_strv_builder_end(builder);
 					str = g_strjoinv(", ", strv);
@@ -544,12 +546,12 @@ handle_device(WacomDeviceDatabase *db,
 
 					for (int i = 0; i < npaired; i++) {
 						const WacomStylus *p = paired[i];
-						char *entry = g_strdup_printf(
+						g_autofree char *entry = g_strdup_printf(
 							"[0x%04x, 0x%06x]",
 							libwacom_stylus_get_vendor_id(
 								p),
 							libwacom_stylus_get_id(p));
-						g_strv_builder_take(builder, entry);
+						g_strv_builder_add(builder, entry);
 					}
 					strv = g_strv_builder_end(builder);
 					str = g_strjoinv(", ", strv);

